### PR TITLE
feat(Ranger): Use Apache Ranger for ACL when replica performs nfs copy

### PR DESCRIPTION
**Pull request by WHBANG**
### What problem does this PR solve? 

https://github.com/apache/incubator-pegasus/issues/1054

This patch add ACL to the NFS copy of replica.

1. Added `gpid` info to the data structure defined in `nfs.thrift`
2. Perform ACL through the Ranger policy matched by `gpid`
3. The registration of nfs is moved to `replica_stub.cpp`, and the original registration 
    information is retained, which is convenient for testing

### CODE REVIEW

File: src/replica/replica_stub.h [hunk 1]
```diff
@@ -270,6 +274,38 @@ class replica_stub : public serverlet<replica_stub>, public ref_counter
...
+        replica_ptr rep = get_replica(pid);
+
+        if (!rep) {
+            TRespType resp;
+            resp.error = ERR_OBJECT_NOT_FOUND;
+            reply(resp);
+            return false;
+        }
+        dsn::message_ex *msg = reply.response_message();
+        if (!rep->access_controller_allowed(msg, ranger::access_type::kWrite)) {
```
Comment (acelyc111) [inline src/replica/replica_stub.h:294]: According to the function name, we can't infer it is used for write or read RPCs, so it's not enough to judge it's write allowed only.

**Review by acelyc111 — DISMISSED** [review]
Good job, thanks for the contribution！

File: src/replica/replica_stub.h [hunk 2]
```diff
@@ -270,6 +274,39 @@ class replica_stub : public serverlet<replica_stub>, public ref_counter
 
     fs_manager *get_fs_manager() { return &_fs_manager; }
 
+    template <typename TReqType, typename TRespType>
+    bool check_status_and_authz_with_reply(const TReqType &request,
+                                           ::dsn::rpc_replier<TRespType> &reply,
+                                           const ::dsn::ranger::access_type &ac_type)
```
Comment (empiredan) [inline src/replica/replica_stub.h:280]:
```suggestion
                                           const ::dsn::ranger::access_type &ac_type) const
```

File: src/replica/replica_stub.h [hunk 3]
```diff
@@ -270,6 +274,39 @@ class replica_stub : public serverlet<replica_stub>, public ref_counter
...
+    {
+        if (!_access_controller->is_enable_ranger_acl()) {
```
Comment (empiredan) [inline src/replica/replica_stub.h:282]: `is_enable_ranger_acl()` could also be declared as const ?

Comment (acelyc111) [reply src/replica/replica_stub.h:282]: It is const.

**Review by acelyc111 — APPROVED** [review]

**Review by empiredan — APPROVED** [review]

**Pull request merged**